scylla-cdc/src/reader.rs

```diff
@@ -0,0 +1,266 @@
+            start_timestamp,
+            window_size,
+            safety_interval,
+            sleep_interval,
+        );
+
+        Ok(reader)
+    }
+
+    fn get_create_table_query() -> String {
+        format!("CREATE TABLE {}.{} (pk int, t int, v text, s text, PRIMARY KEY (pk, t)) WITH cdc = {{'enabled':true}};",
+                TEST_KEYSPACE,
+                TEST_TABLE
+        )
+    }
+
+    async fn create_test_db(session: &Session) {
+        let mut create_keyspace_query = Query::new(format!(
+            "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class': 'SimpleStrategy', 'replication_factor': 1}};",
+            TEST_KEYSPACE
+        ));
+        create_keyspace_query.set_consistency(Consistency::All);
+        session.query(create_keyspace_query, &[]).await.unwrap();
+
+        // Create test table
+        let create_table_query = get_create_table_query();
+        session.query(create_table_query, &[]).await.unwrap();
+    }
+
+    async fn populate_test_db(session: &Session) {
+        session
+            .query(format!("use {};", TEST_KEYSPACE), ())
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 0, 'val0', 'static0');",
+                    TEST_TABLE
+                ),
+                (),
+            )
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 1, 'val1', 'static1');",
+                    TEST_TABLE
+                ),
+                (),
+            )
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 2, 'val2', 'static2');",
+                    TEST_TABLE
+                ),
+                (),
+            )
+            .await
+            .unwrap();
+    }
+
+    async fn get_cdc_stream_id(session: &Session) -> Result<StreamID, Box<dyn Error>> {
+        let query_stream_id = format!(
+            "SELECT \"cdc$stream_id\" FROM {}_scylla_cdc_log limit 1;",
+            TEST_TABLE
+        );
+
+        let mut stream_id_vec: StreamID = vec![];
+        let mut rows = session
+            .query_iter(query_stream_id, &[])
+            .await?
+            .into_typed::<(StreamID,)>();
+
+        while let Some(next_row) = rows.next().await {
+            stream_id_vec = next_row.unwrap().0;
+        }
+        Ok(stream_id_vec)
+    }
+
+    async fn delete_test_data_in_db(session: &Session) {
+        session
+            .query(format!("DROP KEYSPACE IF EXISTS {}", TEST_KEYSPACE), ())
+            .await
+            .unwrap();
+    }
+
+    #[tokio::test]
+    async fn fetch_cdc_data() -> Result<(), Box<dyn std::error::Error>> {
+        let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "172.17.0.2:9042".to_string());
```

Comment:
Can we standardise the default value for this URI or not use any at all? @macher259 uses different one.